plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -375,7 +379,10 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
-        return getColumns(icebergTable.schema(), typeManager).stream()
+        List<IcebergColumnHandle> columns = new ArrayList<>(getColumns(icebergTable.schema(), typeManager));
+        // add hidden path column
+        columns.add(getPathColumnHandle(columns.get(columns.size()-1).getId() + 1));
```

Comment:
I have some doubts regarding the association  of this column id on the fly. Why not using a statically defined negative value for $path, and other syntethic columns that may appear along the way?